src/providers/java.rs

```diff
@@ -95,11 +96,20 @@ impl JavaProvider {
         }
     }
 
+    fn get_gradle_build_cmd(&self, build_gradle_content: &String, gradle_exe: &String) -> String {
+        if self.is_using_spring_boot(build_gradle_content) {
```

Comment:
Is it possible to detect if Spring v1 is used and only use `bootJar` if it is version 2+?

Reply:
One can read it from the `build.gradle` file, however I would not go that way. There can be edge cases, where a gradle plugin update or a spring version update changes the versioning scheme. Also one can define it in kotlin and groovy. That makes capturing this a little bit finnicky and I would not want to rely on a bunch of regexp expressions.

If `nixpacks` would like to support spring v1 as well, there is an alternative solution to what I've provided in this PR. Which is the following:

* The problem is that a `*plain.jar` is being generated along with an executable `.jar`
* The start command can point to the `*plain.jar` instead of the executable
* As a solution one can adjust the start command with the following one-liner: 
`java $JAVA_OPTS -jar -Dserver.port=$PORT $(ls -1 build/libs/*jar | grep -v plain)`
* So basically instead of configuring gradle to create only an executable jar, we use `ls` and `grep` to find the executable `jar` instead of the `plain` one

What do you think about this approach?

Reply:
I think that solution works! And yeah, supporting v1 isn't 100% necessary, but backwards compatibility is always nice, especially if the solution is as simple as greping for the executable jar

Reply:
I've updated the PR with the new solution

Reply:
Is it possible for me to get a review here? And then I can check whether all the pipelines are green or not. Thank you!